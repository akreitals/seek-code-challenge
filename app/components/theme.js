const maxContainerWidth = '940px';

const emoji = '"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol"';
const font = `Averta, -apple-system, BlinkMacSystemFont, “Segoe UI”, Roboto, Helvetica, Arial, sans-serif, ${emoji}`;
const mono = 'SFMono-Regular,"Roboto Mono",Menlo,monospace';

const fontSizes = [12, 14, 16, 18, 24, 32, 48, 64, 72, 96];
const space = [0, 4, 8, 16, 32, 64, 128];
const regular = 400;
const bold = 600;

const greys = {
    black: '#1c1c1c',
    charcoal: '#404040',
    midGrayDark: '#747474',
    midGrayMedium: '#898989',
    midGray: '#a8a8a8',
    midGrayLight: '#d6d6d6',
    grayMedium: '#e1e1e1',
    grayLight: '#eeeeee',
    grayLightest: '#f7f7f7',
    offWhite: '#fbfbfb',
    white: '#ffffff'
};

const brand = {
    blueDarker: '#071d40',
    blueDark: '#042763',
    blue: '#0d3880',
    blueLight: '#184da8',
    blueLighter: '#2765cf',
    pink: '#e60278',
    green: '#157e00',
    greenLight: '#5bc252',
    purple: '#9556b7',
    teal: '#00a5ad',
    orange: '#f57c00',
    yellow: '#ffc600',
    yellowLight: '#fffce3',

    primary: '#e60278',
    accent: '#9556b7',
    success: '#157e00',
    info: '#2765cf',
    warning: '#f57c00',
    error: '#e60278'
};

export const colors = {
    ...greys,
    ...brand
};

export const radius = '2px';

// boxShadows
export const boxShadows = [
    `0 0 2px 0 rgba(0,0,0,.08),0 1px 4px 0 rgba(0,0,0,.16)`,
    `0 0 2px 0 rgba(0,0,0,.08),0 2px 8px 0 rgba(0,0,0,.16)`,
    `0 0 2px 0 rgba(0,0,0,.08),0 4px 16px 0 rgba(0,0,0,.16)`,
    `0 0 2px 0 rgba(0,0,0,.08),0 8px 32px 0 rgba(0,0,0,.16)`
];

const theme = {
    colors,
    font,
    mono,
    fontSizes,
    space,
    regular,
    bold,
    maxContainerWidth,
    radius,
    boxShadows
};

export default theme;
