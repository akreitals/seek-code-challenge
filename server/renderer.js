import React from 'react';
import ReactDomServer from 'react-dom/server';
import { ServerStyleSheet } from 'styled-components';
import { Provider } from 'react-redux';
import { StaticRouter } from 'react-router-dom';
import { configureStore } from '../app/state/configureStore';
import ThemeProvider from '../app/components/ThemeProvider/ThemeProvider';

import RoutesContainer from '../app/containers/Routes/RoutesContainer';

const isProduction = process.env.NODE_ENV === 'production' || false;

const render = (html, styles) => {
    const assetsManifest =
        process.env.webpackAssets && JSON.parse(process.env.webpackAssets);

    return `
    <!DOCTYPE html>
    <html lang="en">
        <head>
            <meta charset="UTF-8">
            <meta name="viewport" content="width=device-width, initial-scale=1">
            <meta http-equiv="x-ua-compatible" content="ie=edge">
            <title>Seek Coding Challenge</title>
            ${styles}
        </head>
        <body>
            <div id="root">${html}</div>
            <script src='${
                isProduction ? assetsManifest['/vendor.js'] : 'js/vendor.js'
            }'></script>
            <script src='${
                isProduction ? assetsManifest['/app.js'] : 'js/app.js'
            }'></script>
        </body>
    </html>
    `;
};

const serverSideRender = (req, res) => {
    const context = {};
    const sheet = new ServerStyleSheet();
    const store = configureStore();

    const appWithRouter = (
        <Provider store={store}>
            <ThemeProvider>
                <StaticRouter location={req.url} context={context}>
                    <RoutesContainer />
                </StaticRouter>
            </ThemeProvider>
        </Provider>
    );

    if (context.url) {
        return res.redirect(context.url);
    }

    const html = ReactDomServer.renderToString(
        sheet.collectStyles(appWithRouter)
    );

    const styles = sheet.getStyleTags();

    return res.status(200).send(render(html, styles));
};

export default serverSideRender;
