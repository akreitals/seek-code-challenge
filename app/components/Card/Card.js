import styled from 'styled-components';
import PropTypes from 'prop-types';
import { borderRadius } from 'styled-system';
import Box from '../Box/Box';
import theme, { colors, boxShadows } from '../theme';

const boxShadow = props => {
    const boxShadowSizes = {
        sm: boxShadows[0],
        md: boxShadows[1],
        lg: boxShadows[2],
        xl: boxShadows[3]
    };

    return {
        boxShadow: boxShadowSizes[props.boxShadowSize]
    };
};

const boxBorder = props => ({
    border:
        props.borderWidth > 0
            ? `${props.borderWidth}px solid ${
                  props.theme.colors[props.borderColor]
              }`
            : null
});

const Card = styled(Box)`
    background-color: ${colors.white};
    ${boxShadow}
    ${boxBorder}
    ${borderRadius}
`;

Card.displayName = 'Card';

Card.propTypes = {
    boxShadowSize: PropTypes.oneOf([null, 'sm', 'md', 'lg', 'xl']),
    borderColor: PropTypes.string,
    ...borderRadius.PropTypes,
    borderWidth: PropTypes.oneOf([0, 1, 2])
};

Card.defaultProps = {
    theme,
    borderColor: null,
    borderRadius: 0,
    borderWidth: 1
};

export default Card;
