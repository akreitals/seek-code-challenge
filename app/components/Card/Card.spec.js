import React from 'react';
import renderer from 'react-test-renderer';
import Card from './Card';
import 'jest-styled-components';
import { colors, boxShadows } from '../theme';

describe('Card', () => {
    test('should render a card with  small box shadow', () => {
        const json = renderer.create(<Card boxShadowSize="sm" />).toJSON();
        expect(json).toMatchSnapshot();
        expect(json).toHaveStyleRule('box-shadow', boxShadows[0]);
        expect(json).toHaveStyleRule('border-radius', '0px');
    });

    test('should render a card with a large shadow', () => {
        const json = renderer.create(<Card boxShadowSize="lg" />).toJSON();
        expect(json).toMatchSnapshot();
        expect(json).toHaveStyleRule('box-shadow', boxShadows[2]);
    });

    test('should render a card with a 1px border', () => {
        const json = renderer.create(<Card borderWidth={1} />).toJSON();
        expect(json).toMatchSnapshot();
        expect(json).toHaveStyleRule('border', `1px solid ${colors.smoke}`);
    });
});
