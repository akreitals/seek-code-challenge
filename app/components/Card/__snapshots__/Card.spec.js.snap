// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should render a card with  small box shadow 1`] = `
.c0 {
  background-color: #ffffff;
  box-shadow: 0 0 2px 0 rgba(0,0,0,.08),0 1px 4px 0 rgba(0,0,0,.16);
  border: 1px solid undefined;
  border-radius: 0px;
}

<div
  className="c0 "
/>
`;

exports[`Card should render a card with a 1px border 1`] = `
.c0 {
  background-color: #ffffff;
  border: 1px solid undefined;
  border-radius: 0px;
}

<div
  className="c0 "
/>
`;

exports[`Card should render a card with a large shadow 1`] = `
.c0 {
  background-color: #ffffff;
  box-shadow: 0 0 2px 0 rgba(0,0,0,.08),0 4px 16px 0 rgba(0,0,0,.16);
  border: 1px solid undefined;
  border-radius: 0px;
}

<div
  className="c0 "
/>
`;
